tests/functional/test_install_extras.py

```diff
@@ -44,12 +44,8 @@ def test_no_extras_uninstall(script):
     result = script.pip(
         'install', 'Paste[openid]==1.7.5.1', expect_stderr=True,
     )
-    assert join(script.site_packages, 'paste') in result.files_created, (
-        sorted(result.files_created.keys())
-    )
-    assert join(script.site_packages, 'openid') in result.files_created, (
-        sorted(result.files_created.keys())
-    )
+    result.did_create(join(script.site_packages, 'paste'))
+    result.did_create(join(script.site_packages, 'openid'))
```

Comment:
We also need to include `sorted(result.files_created.keys()` while doing the assertion, which is not being done in `did_create`

Reply:
Gah. No.

The information about result.files_created is available in `str(result)`, in a much better form. Adding this as a message is not appropriate -- message should be a human readable string describing the error.

Instances like these, where not-all-the-relevant-info messages are provided through the assertion are spread across the test suite, and fixing them is the point of this exercise.

Reply:
Aah, I understand now. Apologies for the oversight.